server.js

```diff
@@ -0,0 +1,386 @@
+            error: "resource " + request.url + " does not exist"
+          };
+          response.end(JSON.stringify(responseBody));
+        });
+        // Return 500 if file name does not match element name
+      } else if (
+        "./public" + request.url !==
+        "./public/" + formDataObject["elementName"].toLowerCase() + ".html"
+      ) {
+        console.log(
+          "resource " +
+            request.url +
+            " and element name " +
+            formDataObject["elementName"].toLowerCase() +
+            ".html" +
+            " do not match"
+        );
+        request.on("end", function() {
+          response.writeHead(500, { "Content-Type": "application/json" });
+          let responseBody = {
+            error:
+              "resource " +
+              request.url +
+              " and element name " +
+              formDataObject["elementName"].toLowerCase() +
+              ".html" +
+              " do not match"
+          };
+          response.end(JSON.stringify(responseBody));
+        });
+        // Return 500 if request is incomplete
+      } else if (
+        !formDataObject["elementName"] ||
+        !formDataObject["elementSymbol"] ||
+        !formDataObject["elementAtomicNumber"] ||
+        !formDataObject["elementDescription"]
+      ) {
+        console.log("Missing one or more values");
+        request.on("end", function() {
+          response.writeHead(500, { "Content-Type": "application/json" });
+          let responseBody = {
+            error: "Missing one or more values"
+          };
+          response.end(JSON.stringify(responseBody));
+        });
+        // Return 500 if file name is index or 404
+      } else if (
+        formDataObject["elementName"] === "index" ||
+        formDataObject["elementName"] === "404"
+      ) {
+        console.log("Invalid elementName");
+        request.on("end", function() {
+          response.writeHead(500, { "Content-Type": "application/json" });
+          let responseBody = {
+            error: "Invalid elementName"
+          };
+          response.end(JSON.stringify(responseBody));
+        });
+        // Create new HTML file and populate with POST data
+      } else {
+        let newElement = `<!DOCTYPE html>
+  <html lang="en">
+    <head>
+      <meta charset="UTF-8" />
+      <title>The Elements - ${formDataObject["elementName"]}</title>
+      <link rel="stylesheet" href="/css/styles.css" />
+    </head>
+    <body>
+      <h1>${formDataObject["elementName"]}</h1>
+      <h2>${formDataObject["elementSymbol"]}</h2>
+      <h3>Atomic number ${formDataObject["elementAtomicNumber"]}</h3>
+      <p>
+        ${formDataObject["elementDescription"]}
+      </p>
+      <p><a href="/">back</a></p>
+    </body>
+  </html>
+  `;
+        fs.writeFile(
+          "./public/" + formDataObject["elementName"].toLowerCase() + ".html",
+          newElement,
+          function(error) {
+            if (error) throw error;
```

Comment:
Errors should be handled properly, they should send back a response with a appropriate status code and message.